### PR TITLE
loop conversion #1

### PR DESCRIPTION
**Pull request by ajax16384**

### CODE REVIEW

File: src/torrent.cpp [hunk 1]
```diff
@@ -9586,7 +9558,7 @@ namespace libtorrent
 		// now, iterate over all time critical pieces, in order of importance, and
 		// request them from the peers, in order of responsiveness. i.e. request
 		// the most time critical pieces from the fastest peers.
-		for (std::vector<time_critical_piece>::iterator i = m_time_critical_pieces.begin()
+		for (auto i = m_time_critical_pieces.begin()
```
Comment (arvidn) [inline src/torrent.cpp]: if you would introduce a ``bool first`` variable, I think you could make this a range-for loop too

File: src/torrent.cpp [hunk 2]
```diff
@@ -7690,16 +7665,14 @@ namespace libtorrent
 			if (p.peer_info_struct() && p.peer_info_struct()->seed)
 				++seeds;
 
-			for (std::vector<pending_block>::const_iterator j = p.request_queue().begin()
-				, end(p.request_queue().end()); j != end; ++j)
+			for (auto const& j : p.request_queue())
```
Comment (arvidn) [inline src/torrent.cpp]: there are some conflicts here I believe, with a patch that I have in flight. and I'm afraid I intend to land mine first

File: src/torrent.cpp [hunk 3]
```diff
@@ -3419,11 +3411,10 @@ namespace libtorrent
 		std::vector<file_slice> files = fs.map_block(
 			p.piece_index, offset, (std::min)(piece_size - offset, block_size()));
 		std::int64_t ret = 0;
-		for (std::vector<file_slice>::iterator i = files.begin()
-			, end(files.end()); i != end; ++i)
+		for (auto const& i :  files)
```
Comment (arvidn) [inline src/torrent.cpp]: would you mind removing one of the spaces in front of "files"?

File: src/torrent.cpp (hunk 2 above)
Comment (ajax16384) [inline src/torrent.cpp]: ok. i have already noticed your PR and skipped peer_connection

File: src/torrent.cpp (hunk 1 above)
Comment (ajax16384) [inline src/torrent.cpp]: ok

File: src/torrent.cpp (hunk 3 above)
Comment (ajax16384) [inline src/torrent.cpp]: oops. ok

File: src/torrent.cpp (hunk 2 above)
Comment (ajax16384) [inline src/torrent.cpp]: this one https://github.com/arvidn/libtorrent/pull/1782 ?

**Comment by arvidn** [comment]
yes, that was the PR. I just merged it, please rebase

**Comment by ajax16384** [comment]
ok

File: src/torrent.cpp [hunk 4]
```diff
@@ -9515,8 +9492,8 @@ namespace libtorrent
 		// now, iterate over all time critical pieces, in order of importance, and
 		// request them from the peers, in order of responsiveness. i.e. request
 		// the most time critical pieces from the fastest peers.
-		for (std::vector<time_critical_piece>::iterator i = m_time_critical_pieces.begin()
-			, end(m_time_critical_pieces.end()); i != end; ++i)
+		bool non_first_piece{false};
```
Comment (arvidn) [inline src/torrent.cpp]: this is a bit of a nit-pick, but would you mind negating the name and meaning of this? It just feels like a double-negation right now

**Comment by arvidn** [comment]
lgtm, apart from comment

File: src/torrent.cpp (hunk 4 above)
Comment (ajax16384) [inline src/torrent.cpp]: ok

**Comment by arvidn** [comment]
lgtm

**Pull request merged**